crates/oxc_linter/src/rules/typescript/no_dynamic_delete.rs

```diff
@@ -0,0 +1,162 @@
+use oxc_ast::{ast::Expression, AstKind};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+use oxc_syntax::operator::UnaryOperator;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Default, Clone)]
+pub struct NoDynamicDelete;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Disallow using the delete operator on computed key expressions.
+    ///
+    /// ### Why is this bad?
+    /// Deleting dynamically computed keys can be dangerous and in some cases not well optimized.
+    /// Using the delete operator on keys that aren't runtime constants could be a sign that you're using the wrong data structures.
+    /// Consider using a Map or Set if you’re using an object as a key-value collection.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// const container: { [i: string]: 0 } = {};
+    /// delete container['aa' + 'b'];
+    /// ```
+    NoDynamicDelete,
+    restriction,
+);
+
+fn no_dynamic_delete_diagnostic(span0: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn(
+        "typescript-eslint(no-dynamic-delete): Do not delete dynamically computed property keys.",
+    )
+    .with_help("Disallow using the `delete` operator on computed key expressions")
+    .with_label(span0)
+}
+
+impl Rule for NoDynamicDelete {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::UnaryExpression(expr) = node.kind() else { return };
+        if expr.operator != UnaryOperator::Delete {
```

Comment:
prefer `matches!` to remove a function call dispatch (`matches!` is a const macro`)

Reply:
fixed it!